### ISSUE TITLE
Crash when app launch

### ISSUE THREAD

**Issue by vitonzhangtt** [firebase/firebase-ios-sdk#6199]
### Step 0: Are you in the right place?
I think that crash is related to firebase SDK.
I found this issue https://github.com/firebase/firebase-ios-sdk/issues/3814 which was closed without being resolved.
 
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.
  * Firebase SDK version: 6.22.0
  * Firebase Component: Core, Performance 
  * Component version: 6.22.0
  * Installation method: `CocoaPods` 

#### Steps to reproduce:

What happened? How can we make the problem occur?
App crash when app is launching.  
The crash type is 0x8badf00d. 
This crash is random. 
We have no specific steps to reproduce this crash.

#### Relevant Code:
The stack of the crashlog is as the following. I replace my application name with "SomeApp" in the crashlog.

```
{"app_name":"SomeApp","timestamp":"2020-07-30 10:07:01.00 -0700","app_version":"4.1.6","slice_uuid":"4edbf132-ab5d-370e-937b-d785bb07e7ff","adam_id":861891048,"build_version":"4.1.6.1","bundleID":"com.labs. SomeApp","share_with_app_devs":0,"is_first_party":0,"bug_type":"109","os_version":"iPhone OS 13.6 (17G68)","incident_id":"52740B99-979E-4DCF-B981-0727F8F7EDBA","name":"SomeApp"}
Incident Identifier: 52740B99-979E-4DCF-B981-0727F8F7EDBA
CrashReporter Key:   6997e226a148102f4741b2b25d2322fa5471680a
Hardware Model:      iPad11,3
Process:             SomeApp [2906]
Path:                /private/var/containers/Bundle/Application/61091226-C872-411B-9902-E53A1938C814/SomeApp.app/SomeApp
Identifier:          com.labs.SomeApp
Version:             4.1.6.1 (4.1.6)
AppStoreTools:       11E707
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           com.labs.SomeApp [1394]


Date/Time:           2020-07-30 10:07:00.5388 -0700
Launch Time:         2020-07-30 10:06:40.3421 -0700
OS Version:          iPhone OS 13.6 (17G68)
Release Type:        User
Baseband Version:    n/a
Report Version:      104

Exception Type:  EXC_CRASH (SIGKILL)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: Namespace SPRINGBOARD, Code 0x8badf00d
Termination Description: SPRINGBOARD, scene-create watchdog transgression: application<com.labs.SomeApp >:2906 exhausted real (wall clock) time allowance of 18.47 seconds | ProcessVisibility: Foreground | ProcessState: Running | WatchdogEvent: scene-create | WatchdogVisibility: Foreground | WatchdogCPUStatistics: ( | "Elapsed total CPU time (seconds): 33.280 (user 33.280, system 0.000), 28% CPU", | "Elapsed application CPU time (seconds): 0.559, 0% CPU" | )
Triggered by Thread:  0

Thread 0 name:  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x00000001992a37e4 __ulock_wait + 8
1   libsystem_platform.dylib      	0x00000001991bd000 _os_unfair_lock_lock_slow + 196
2   libobjc.A.dylib               	0x00000001991e0464 lookUpImpOrForward + 136
3   libobjc.A.dylib               	0x00000001991cce20 _objc_msgSend_uncached + 64
4   SomeApp                        	0x0000000101500104 -[CLSCrashReportingController installCrashReportingHandlers:] + 44
5   SomeApp                        	0x00000001014ff3dc -[CLSCrashReportingController startCrashReporterWithAPIKey:betaToken:profilingMark:report:] + 620
6   SomeApp                        	0x00000001014fefdc -[CLSCrashReportingController startWithProfilingMark:betaToken:] + 304
7   SomeApp                        	0x00000001014f3fa0 __20-[Crashlytics start]_block_invoke + 628
8   libdispatch.dylib             	0x000000019915d5ac _dispatch_client_callout + 20
9   libdispatch.dylib             	0x000000019915ecd0 _dispatch_once_callout + 32
10  SomeApp                        	0x00000001014f3d28 -[Crashlytics start] + 156
11  SomeApp                        	0x00000001014f414c +[Crashlytics initializeIfNeeded] + 48
12  SomeApp                        	0x00000001019c3f60 __15+[Fabric with:]_block_invoke + 524
13  libdispatch.dylib             	0x000000019915d5ac _dispatch_client_callout + 20
14  libdispatch.dylib             	0x000000019915ecd0 _dispatch_once_callout + 32
15  SomeApp                        	0x00000001019c3d4c +[Fabric with:] + 228
16  SomeApp                        	0x0000000104a7ed0c -[XXXBaseLibModule appManager:willFinishLaunchingWithOptions:] + 74444044 (XXXBaseLibModule.m:0)
17  SomeApp                        	0x0000000103051940 __60-[XXXAppManager application:willFinishLaunchingWithOptions:]_block_invoke + 46995776 (XXXAppManager.m:123)
18  SomeApp                        	0x00000001030517f8 __41-[XXXAppManager performSelector:hanlder:]_block_invoke + 46995448 (XXXAppManager.m:112)
19  CoreFoundation                	0x000000019948d71c __NSARRAY_IS_CALLING_OUT_TO_A_BLOCK__ + 24
20  CoreFoundation                	0x000000019938b978 -[__NSArrayM enumerateObjectsWithOptions:usingBlock:] + 196
21  SomeApp                        	0x0000000103051780 -[XXXAppManager performSelector:hanlder:] + 46995328 (XXXAppManager.m:113)
22  SomeApp                        	0x00000001030518b4 -[XXXAppManager application:willFinishLaunchingWithOptions:] + 46995636 (XXXAppManager.m:125)
23  SomeApp                        	0x0000000104a7d3f8 -[XXXAppDelegate application:willFinishLaunchingWithOptions:] + 74437624 (AppDelegate.m:84)
24  UIKitCore                     	0x000000019d5d9d4c -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 216
25  UIKitCore                     	0x000000019d5dbbbc -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] + 5104
26  UIKitCore                     	0x000000019d5e1588 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1256
27  UIKitCore                     	0x000000019cd51d60 -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] + 152
28  UIKitCore                     	0x000000019d21d588 _UIScenePerformActionsWithLifecycleActionMask + 112
29  UIKitCore                     	0x000000019cd52894 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke + 212
30  UIKitCore                     	0x000000019cd522c0 -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] + 296
31  UIKitCore                     	0x000000019cd526b0 -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 756
32  UIKitCore                     	0x000000019cd51f38 -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] + 340
33  UIKitCore                     	0x000000019cd56654 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke_2 + 196
34  UIKitCore                     	0x000000019d13bf8c +[BSAnimationSettings+ 6057868 (UIKit) tryAnimatingWithSettings:actions:completion:] + 856
35  UIKitCore                     	0x000000019d23722c _UISceneSettingsDiffActionPerformChangesWithTransitionContext + 260
36  UIKitCore                     	0x000000019cd56390 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke + 156
37  UIKitCore                     	0x000000019d237114 _UISceneSettingsDiffActionPerformActionsWithDelayForTransitionContext + 108
38  UIKitCore                     	0x000000019cd561e8 -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] + 392
39  UIKitCore                     	0x000000019cbbc918 __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke + 640
40  UIKitCore                     	0x000000019cbbb418 -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] + 256
41  UIKitCore                     	0x000000019cbbc648 -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] + 236
42  UIKitCore                     	0x000000019d5df828 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 552
43  UIKitCore                     	0x000000019d162780 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 376
44  FrontBoardServices            	0x000000019e6fe084 -[FBSSceneImpl _callOutQueue_agent_didCreateWithTransitionContext:completion:] + 440
45  FrontBoardServices            	0x000000019e724610 __86-[FBSWorkspaceScenesClient sceneID:createWithParameters:transitionContext:completion:]_block_invoke.168 + 116
46  FrontBoardServices            	0x000000019e708f40 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240
47  FrontBoardServices            	0x000000019e7242a4 __86-[FBSWorkspaceScenesClient sceneID:createWithParameters:transitionContext:completion:]_block_invoke + 344
48  libdispatch.dylib             	0x000000019915d5ac _dispatch_client_callout + 20
49  libdispatch.dylib             	0x000000019916033c _dispatch_block_invoke_direct + 264
50  FrontBoardServices            	0x000000019e74a6c8 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48
51  FrontBoardServices            	0x000000019e74a374 -[FBSSerialQueue _queue_performNextIfPossible] + 432
52  FrontBoardServices            	0x000000019e74a8e0 -[FBSSerialQueue _performNextFromRunLoopSource] + 32
53  CoreFoundation                	0x0000000199436d14 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
54  CoreFoundation                	0x0000000199436c68 __CFRunLoopDoSource0 + 84
55  CoreFoundation                	0x00000001994363c4 __CFRunLoopDoSources0 + 188
56  CoreFoundation                	0x00000001994311fc __CFRunLoopRun + 780
57  CoreFoundation                	0x0000000199430bc8 CFRunLoopRunSpecific + 480
58  GraphicsServices              	0x00000001a38125cc GSEventRunModal + 164
59  UIKitCore                     	0x000000019d5e3744 UIApplicationMain + 1936
60  SomeApp                        	0x0000000104a6b5e4 main + 74364388 (main.m:43)
61  libdyld.dylib                 	0x00000001992ad384 start + 4

```

**Comment by maksymmalyhin** [comment]
@vitonzhangtt Thank you for the report. It looks the main thread of your app is blocked by a deprecated version of Fabric Crashlytics. You should upgrade Crashlytics  in your app following the [guide](https://firebase.google.com/docs/crashlytics/upgrade-sdk?platform=ios). More information of integration with Crashlytics and Fabric deprecation can be found [here](https://firebase.google.com/docs/crashlytics/get-started?platform=ios). 

Please let us know if it helped. If you have another issue with updated Crashlytics SDK you most likely will need to create another ticket.
